latest changes - PR instead of rebase

SmarterHomes opened the pull request.

brakmic commented:
taking back the pull request, will add code from "work-on-rpcmisc" to the previous pull-request

UdjinM6 committed:
Split sentinel expiration in CMasternode::Check() in two parts (timeout and version) (#2121)

The initial ping is sent with defaults which switch MNs to SENTINEL_PING_EXPIRED state
while they should really be in PRE_ENABLED state. The fix is to split this verification
in two parts - this way sentinel version is only checked after at least one ping is received
from the masternode itself and not from the cold wallet.

UdjinM6 committed:
Do not create mnb until sync is finished (#2122)

We don't relay mnb if sync is still in progress. Creating mnb without
relaying it only increases inconsistency and leads to user confusion.
Unifying conditions for creation and relaying fixes this.

UdjinM6 committed:
Don't drop mnb-s for outdated MNs (#2131)

Instead just mark them as MASTERNODE_UPDATE_REQUIRED and proceed further.

InhumanPerfection committed:
Fix an edge case in PrepareDenominate (#2138)

* PrepareDenominate fix

* Fix conditions

UdjinM6 committed:
Fix block value/payee validation in lite mode (#2148)

UdjinM6 committed:
Use correct protocol when serializing messages in reply to `getdata` (#2157)

Messages should be serialized according the protocol of the peer who asked us or otherwise peers running on other protocols won't be able to deserialize the message correctly.

Commit:
update release notes
